ios/FluentUI/AvatarGroup/AvatarGroup.swift

```diff
@@ -145,16 +189,26 @@ public struct AvatarGroup: View {
     /// - Parameters:
     ///   - style: The style of the avatar group.
     ///   - size: The size of the avatars displayed in the avatar group.
-    init(style: MSFAvatarGroupStyle,
-         size: MSFAvatarSize) {
+    public init(style: MSFAvatarGroupStyle,
+                size: MSFAvatarSize,
+                avatarCount: Int,
+                @ViewBuilder avatarAtIndex: @escaping (_ index: Int) -> Avatar) {
         let state = MSFAvatarGroupStateImpl(style: style,
-                                            size: size)
+                                            size: size,
+                                            avatarCount: avatarCount,
+                                            avatarAtIndex: avatarAtIndex)
         self.state = state
         self.tokens = state.tokens
     }
 
+    var avatars: [Avatar] {
+        return (0..<state.avatarCount).map { index in
+            self.state.avatarAtIndex(index)
+        }
+    }
+
     public var body: some View {
-        let avatars: [MSFAvatarStateImpl] = state.avatars
+        let avatars = self.avatars
```

Comment:
we need this?

Reply:
Yep! `self.avatars` is a calculated property, so it'll generate the list of `Avatar` views on every call. We reference this array several times in this property, so it's important to cache it locally, and I intentionally used the same variable name so we can't accidentally fall back on the generators again by looking for an array called `avatars` (that'll now default to the local one; only an explicit `self.avatars` will trigger regeneration.